pkg/lib/message.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package lib
+
+import (
+	"bytes"
+	"errors"
+	"github.com/moov-io/iso8583/pkg/utils"
+)
+
+type Iso8583Message interface {
+	Bytes() ([]byte, error)
+	Load(raw []byte) (int, error)
+	Validate() error
+	GetElements() map[int]*Element
+	GetMti() *Element
+	GetBitmap() *Element
+}
+
+// create data elements of message with specification
+func NewMessage(spec *utils.Specification) (Iso8583Message, error) {
+	if spec == nil && spec.Elements == nil || spec.Encoding == nil {
+		return nil, errors.New("has invalid specification")
+	}
+	_elements, err := NewDataElements(spec)
+	if err != nil {
+		return nil, err
+	}
+	return &isoMessage{
+		Mti: &Element{
+			Type:     utils.ElementTypeMti,
+			Fixed:    true,
+			Length:   4,
+			Encoding: spec.Encoding.MtiEnc,
+		},
+		Bitmap: &Element{
+			Type:     utils.ElementTypeBitmap,
+			Fixed:    true,
+			Length:   64,
+			Encoding: spec.Encoding.BitmapEnc,
+		},
+		Elements: _elements,
+		Spec:     spec,
+	}, nil
+}
+
+// message instance
+// isoMessage is structure for ISO 8583 message encode and decode
+type isoMessage struct {
+	Mti          *Element             `xml:"mti,omitempty" json:"mti,omitempty" yaml:"mti,omitempty"`
```

Comment:
@adamdecaf this is a MTI and acrynom. Should it all be capitalized?

Reply:
Yea, if we can.